### PR TITLE
Staging sites: improve the error message when a permanent sync error occurs

### PR DESCRIPTION
**Pull request by sejas**
- Closes https://github.com/Automattic/dotcom-forge/issues/4201

## Proposed Changes

* Display two types of error. We will display a more permanent error if we receive from the API that a site (pro or staging) cannot sync `!can_sync`.

## Disconnecting Staging

* Create an atomic site and a staging site
* On the staging site, access `wp-cli` and run `wp jetpack disconnect blog`
* Open Settings > Hosting Configuration on production using this branch or calypso live
* Try to sync pull/push.
* Observe the message indicates the staging url and ask to contact support.
* Refresh the page and try the other option pull/pussh.
* Observe the same error message

## Disconnecting Production

* Run the same steps as before, but running the `wp jetpack disconnect blog` on production, and accessing Hosting Configuration on staging.

## Screenshots

| **Before** | **After** |
| ---------- | --------- |
| Screenshot 2023-10-19 at 19 29 59 |  Screenshot 2023-10-19 at 19 33 44 |

https://github.com/Automattic/wp-calypso/assets/779993/e6adade9-61d6-47b5-9ce3-0c04b87e617b

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by sejas** [comment]
@wojtekn , are these messages what we are looking for?

Adding the site url makes more clear what site is failing.

File: client/my-sites/hosting/staging-site-card/card-content/staging-sync-card.tsx [hunk 1]
```diff
@@ -302,7 +319,25 @@ const SyncCardContainer = ( {
...
-					{ error && (
+					{ error && isStagingSyncError && (
+						<Notice
+							status="is-error"
+							icon="mention"
+							showDismiss={ false }
+							text={ translate(
+								'We couldn’t connect to the %(siteType)s site: {{br/}} %(siteUrl)s',
+								{
+									args: { siteType: siteToSync, siteUrl: siteUrls[ siteToSync ] ?? '' },
```
Comment (wojtekn) [inline client/my-sites/hosting/staging-site-card/card-content/staging-sync-card.tsx]: Would it make sense to use `urlToSlug()` to remove the protocol part?

**Review by wojtekn — APPROVED** [review]
@sejas I tested the workflow, and the user experience is better now. Before, I got a cryptic error and "Try again" CTA, which didn't help solve the issue.

Now, I have clear information about the broken site, and CTA is asking me to contact support. I simulated getting with support and fixed the Jetpack connection. Then, I was able to refresh the page and schedule synchronization.

File: client/my-sites/hosting/staging-site-card/card-content/staging-sync-card.tsx (hunk 1 above)
Comment (sejas) [reply client/my-sites/hosting/staging-site-card/card-content/staging-sync-card.tsx]: I think it's a great idea.

Added here: 
https://github.com/Automattic/wp-calypso/pull/83259/commits/0da0fd4db9c6a7d78bb2d2be08a9872bae481cdb

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9486417

Some locales (Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @sejas for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9486417).

**Pull request merged**